### ISSUE TITLE
apollo-composable "import not found: DefaultApolloClient"

### ISSUE THREAD

**Issue by longfellowone** [vuejs/vue-apollo#1029]
**Describe the bug**
App does not load "import not found: DefaultApolloClient"

**To Reproduce**
1. git clone https://github.com/longfellowone/todo-vue.git
2. npm run dev

**Expected behavior**
No error

**Versions**
vue: 3.0.0-rc.5
apollo-composable: 4.0.0-alpha.10
apollo-boost: 0.4.9

**Additional context**
No other imports working from apollo-composable (useQuery etc.)

**Comment by pepsighan** [comment]
I think the problem is with vite. I am using a similar setup and its not working.

**Comment by justinwaite** [comment]
Seeing this as well. As a temporary workaround, you can import the default export like so:
```
import apollo from '@vue/apollo-composable` 
```
and then use the other exports from there:
```
  app.provide(apollo.DefaultApolloClient, apolloClient)
```

**Comment by pepsighan** [comment]
@justinwaite I tried the same thing but encountered the same error.

**Comment by lzurbriggen** [comment]
@pepsighan Yes, this seems to be an Issue with either `vite` or the packaging of the library. The imports work fine in a webpack project.

@justinwaite There are 2 changes I had to make to get queries working, the first one is the error you mentioned:
1. may be an issue with either `vue-demi` or more likely an API change in a newer version of the composition-api. Issue is [this line](https://github.com/vuejs/vue-apollo/blob/1e7a5c6a3b5d40fe800d104d2235aa6b121d2cb3/packages/vue-apollo-composable/src/util/loadingTracking.ts#L14): `const root: any = getCurrentInstance().$root`. I temporarily fixed this by replacing `$root` with `root`.
1. the second issue occurs on [this line](https://github.com/vuejs/vue-apollo/blob/1e7a5c6a3b5d40fe800d104d2235aa6b121d2cb3/packages/vue-apollo-composable/src/useQuery.ts#L140) on `onServerPrefetch`. Temporarily fixed by removing the whole function call (I don't use SSR)

I believe this is related to #1011

**Comment by longfellowone** [comment]
If looking for a fix to 

```Uncaught (in promise) TypeError: Cannot read property '_apolloAppTracking' of undefined```

See comment below from #1011 
https://github.com/vuejs/vue-apollo/issues/1011#issuecomment-674873671

**Comment by jessielaf** [comment]
Hi I am running into this issue with nuxt on 4.0.0-alpha.10. I cannot use another version because of #1081. Is there any work around?

**Comment by jsbroks** [comment]
```
v1.0.0-beta.22
@github-actions github-actions released this 26 days ago · 11 commits to master since this release

Features
getCurrentInstance: Aligning with vue3 (#520) (1495a46)
BREAKING CHANGES
getCurrentInstance: The internal vm can be accessed with getCurrentInstance().proxy
```
So it needs to be `getCurrentInstance().proxy.$root`

**Comment by madmod** [comment]
The following vite plugin worked for me:
```javascript
    {
      name: 'fix-apollo-composable',
      enforce: 'post',
      transform(src, id) {
        if (id.includes('@vue/apollo-composable/dist/useQuery.js')) {
          return {
            code: src
              .replace(/onServerPrefetch,/, '')
              .replace(/\:\sonServerPrefetch\(/, ': (')
              .replace(/onServerPrefetch/, 'null')
              .replace(/(.* require\("vue"\);)/m, ''),
            id,
          }
        }
      },
    },
```

*Edit*
I also had to add `@vue/apollo-composable` to `optimizeDeps.exclude` in order for the dev build to work.

**Comment by abdullahseba** [comment]
How come this is taking so long to fix?  The example in the docs causes this error so it's not an edge case!

https://v4.apollo.vuejs.org/guide-composable/query.html#usequery
